x-pack/plugins/alerting/server/alerts_service/lib/set_alerts_to_untracked.ts

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { Logger } from '@kbn/logging';
+import {
+  ALERT_RULE_CONSUMER,
+  ALERT_RULE_TYPE_ID,
+  ALERT_RULE_UUID,
+  ALERT_STATUS,
+  ALERT_STATUS_ACTIVE,
+  ALERT_STATUS_UNTRACKED,
+  ALERT_UUID,
+} from '@kbn/rule-data-utils';
+
+export interface SetAlertsToUntrackedOpts {
+  indices: string[];
+  ruleIds?: string[];
+  alertUuids?: string[];
+  ensureAuthorized?: (opts: { ruleTypeId: string; consumer: string }) => Promise<unknown>;
+}
+
+type UntrackedAlertsResult = Array<{ [ALERT_RULE_UUID]: string; [ALERT_UUID]: string }>;
+interface ConsumersAndRuleTypesAggregation {
+  ruleTypeIds: {
+    buckets: Array<{
+      key: string;
+      consumers: {
+        buckets: Array<{ key: string }>;
+      };
+    }>;
+  };
+}
+
+export async function setAlertsToUntracked({
+  logger,
+  esClient,
+  indices,
+  ruleIds = [],
+  alertUuids = [], // OPTIONAL - If no alertUuids are passed, untrack ALL ids by default,
+  ensureAuthorized,
+}: {
+  logger: Logger;
+  esClient: ElasticsearchClient;
+} & SetAlertsToUntrackedOpts): Promise<UntrackedAlertsResult> {
+  if (isEmpty(ruleIds) && isEmpty(alertUuids))
+    throw new Error('Must provide either ruleIds or alertUuids');
+
+  const shouldMatchRules: Array<{ term: Record<string, { value: string }> }> = ruleIds.map(
+    (ruleId) => ({
+      term: {
+        [ALERT_RULE_UUID]: { value: ruleId },
+      },
+    })
+  );
+  const shouldMatchAlerts: Array<{ term: Record<string, { value: string }> }> = alertUuids.map(
+    (alertId) => ({
+      term: {
+        [ALERT_UUID]: { value: alertId },
+      },
+    })
+  );
+
+  const statusTerms: Array<{ term: Record<string, { value: string }> }> = [
+    {
+      term: {
+        [ALERT_STATUS]: { value: ALERT_STATUS_ACTIVE },
+      },
+    },
+  ];
+
+  const must = [
+    ...statusTerms,
+    {
+      bool: {
+        should: shouldMatchRules,
+      },
+    },
+    {
+      bool: {
+        should: shouldMatchAlerts,
+        // If this is empty, ES will default to minimum_should_match: 0
+      },
+    },
+  ];
+
+  if (ensureAuthorized) {
```

Comment:
ensureAuthorized should always be there if we can, we need to enforce this check!

Reply:
`setAlertsToUntracked` is sometimes called through `untrackRuleAlerts`, which is only ever called after a different `ensureAuthorized` method has run

Reply:
let's go with the way it is and we can refactor it later on